iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorLinearModelImpl.java

```diff
@@ -79,12 +89,32 @@ public boolean containsSection(int sectionNumber) {
     }
 
     @Override
-    public double getB(int sectionNum) {
-        return bPerSection * sectionNum;
+    public double getB(int sectionIndex) {
+        return sectionIndex == 0 ? 0 : bPerSection;
+    }
+
+    @Override
+    public double getG(int sectionIndex) {
+        return Double.isNaN(gPerSection) || sectionIndex == 0 ? 0 : gPerSection;
+    }
```

Comment:
You should check if the sectionIndex is in the range [0, maxSectionCount]?

Reply:
Done